src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,33 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot save actor: " + actor.getName()
+                    + " to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
```

Comment:
Use try with finally, where you use transaction and try with resources with Read operations. Here, you have correctly used try with finally for the transactional operation in the `add` method.